Sources/AardvarkReveal/RevealAttachmentGenerator.swift

```diff
@@ -0,0 +1,518 @@
+                DispatchQueue.main.async {
+                    delegate?.revealAttachmentGeneratorDidFinishCapturingAppState(success: false)
+                }
+
+                completionQueue.async {
+                    completion(nil)
+                }
+                return
+            }
+
+            DispatchQueue.main.async {
+                delegate?.revealAttachmentGeneratorDidFinishCapturingAppState(success: true)
+            }
+
+            do {
+                let archiveBuilder = ARKRevealArchiveBuilder()
+
+                try self.buildRevealPackage(
+                    from: applicationStateData,
+                    baseRevealURL: baseRevealURL,
+                    urlSession: session,
+                    archiveBuilder: archiveBuilder,
+                    completionQueue: completionQueue
+                ) { success in
+                    guard
+                        success,
+                        let archive = archiveBuilder.completeArchive(),
+                        let compressedArchive = ARKCompressionUtility.gzippedData(for: archive)
+                    else {
+                        DispatchQueue.main.async {
+                            delegate?.revealAttachmentGeneratorDidFinishBundlingRevealFile(success:false)
+                        }
+
+                        completion(nil)
+                        return
+                    }
+
+                    DispatchQueue.main.async {
+                        delegate?.revealAttachmentGeneratorDidFinishBundlingRevealFile(success: true)
+                    }
+
+                    completion(
+                        ARKBugReportAttachment(
+                            fileName: "\(self.applicationName()).reveal.tar.gz",
+                            data: compressedArchive,
+                            dataMIMEType: "application/gzip"
+                        )
+                    )
+                }
+
+            } catch {
+                DispatchQueue.main.async {
+                    delegate?.revealAttachmentGeneratorDidFinishBundlingRevealFile(success: false)
+                }
+
+                completionQueue.async {
+                    completion(nil)
+                }
+            }
+        }
+
+        dataTask.resume()
+    }
+
+    // MARK: - Private Methods
+
+    @available(iOS 11, *)
+    private func buildRevealPackage(
+        from applicationStateData: Data,
+        baseRevealURL: URL,
+        urlSession: URLSession,
+        archiveBuilder: ARKRevealArchiveBuilder,
+        completionQueue: DispatchQueue,
+        completion: @escaping (_ success: Bool) -> Void
+    ) throws {
+        // Create the bundle based on the application name, clamped to a reasonable length so that the file paths don't
+        // go over 100 characters (a limitation of the archive builder).
+        let appName = applicationName()
+        let bundleName = "\(appName.prefix(50)).reveal"
```

Comment:
50 is significantly lower than 100 😅

Reply:
It is indeed. The longest file path right now is for the resource images:

```
<app_name>.reveal/Resources/2692677334741286913#1.png
```

That's the app name, followed by ".reveal/Resources/", a numeric identifier, "#", either "0" or "1", and the file extension. I'm assuming the identifiers are always 19 digits, although I'm not 100% sure of that. That gives us 43 characters plus a null terminator. So in theory this limit could be 56 characters, but 50 felt like a reasonable limit that gives a little bit of padding in case I'm wrong about the length of the identifier being capped at 19.

Reply:
Got it. The file path limit is def odd. How you're tackling this makes sense.